lte/cloud/go/services/subscriberdb/storage/sub_proto.go

```diff
@@ -0,0 +1,335 @@
+			PrimaryKey(subProtoNidCol, subProtoSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store table")
+		}
+		_, err = l.builder.CreateTable(subProtoTmpTableName).
+			IfNotExists().
+			Column(subProtoNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subProtoNidCol, subProtoSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store tmp table")
+		}
+		return nil, nil
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InsertManyByNetwork serializes and inserts a batch of subscriber protos into the temporary table.
+func (l *SubProtoStore) InsertManyByNetwork(network string, subProtos []*lte_protos.SubscriberData) error {
+	if len(subProtos) == 0 {
+		return nil
+	}
+	insertQuery := l.builder.
+		Insert(subProtoTmpTableName).
+		Columns(subProtoNidCol, subProtoSidCol, subProtoProtoCol)
+	errs := &multierror.Error{}
+	for _, subProto := range subProtos {
+		marshaled, err := proto.Marshal(subProto)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "serialize subproto of ID %+v", lte_protos.SidString(subProto.Sid)))
+			continue
+		}
+		insertQuery = insertQuery.Values(network, lte_protos.SidString(subProto.Sid), marshaled)
+	}
+	if errs.ErrorOrNil() != nil {
+		return errs
+	}
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := insertQuery.RunWith(tx).Exec()
+		return nil, errors.Wrapf(err, "insert sub protos into store for network %+v", network)
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// CommitUpdateByNetwork cleans up and re-populates the subscriber proto store table with data from the temporary
+// table for a particular network, and then truncates the temporary table.
+func (l *SubProtoStore) CommitUpdateByNetwork(network string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		// HACK: hard coding sql query because there currently doesn't exist good support
+		// for this following query with squirrel
+		deleteRowsOnlyInRealTableQuery := fmt.Sprintf(
+			"DELETE FROM %s WHERE (%s = \"%s\") AND ((%s, %s) NOT IN (SELECT %s, %s FROM %s))",
+			subProtoTableName, subProtoNidCol, network,
+			subProtoNidCol, subProtoSidCol,
+			subProtoNidCol, subProtoSidCol, subProtoTmpTableName,
+		)
+		_, err := tx.Exec(deleteRowsOnlyInRealTableQuery)
+		if err != nil {
+			return nil, errors.Wrapf(err, "clean up previous sub proto store table")
+		}
+
+		// the upsertColumnPrefix is used to refer to the the table whose column value is
+		// used in the "on conflict update" operation; it depends on the sql dialect
+		// e.g. for mariaDB, "f1=t2.f1"; for postgres, "f1=excluded.f1"
+		var upsertColumnPrefix string
```

Comment:
Move this whole thing (get sql dialect, choose which upsert format string, and actually formatting the string) to sqorc helper fn